### PR TITLE
GUACAMOLE-935: Free internals of rdpBitmap only when required.

### PR DESCRIPTION
**Pull request by mike-jumper**
This change adds an test to the `configure` script which determines whether we are building against FreeRDP 2.0.0-rc0, which had a `Bitmap_Free()` which freed bitmaps. For all newer versions of FreeRDP 2.0.0, `Bitmap_Free()` doesn't - it simply wraps the implementation-specific free handler, and it's on the implementation to know which internal parts of `rdpBitmap` were allocated by `Bitmap_Alloc()` and must now be freed.

The latter was handled in commit 8dda26a when the resulting memory leak was discovered during valgrind testing, but this now must be made conditional as this would result in a double-free against 2.0.0-rc0.

Unfortunately, the test itself is awkward:

* It is not possible to test `Bitmap_Free()` behavior in itself. We must test version numbers.
* While we can test numeric version macros, the only numeric version macros available always indicate that the version is "2.0.0".
* Version macros exist which indicate the RC number and inter-RC number, but they are strings. The C preprocessor cannot test string values.

Because of these limitations, the test can only be performed by either compiling a test program and running it (which can only be done when not cross-compiling) or by checking preprocessor output. This change ultimately goes with the latter approach.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -578,6 +578,30 @@ then
...
+    # associated data member within Bitmap_Free(), relying on the
+    # implementation-specific free handler to free only implementation-specific
+    # data. This changed in commit 2cf10cc, and implementations must now
+    # manually free all data associated with the rdpBitmap, even data which
+    # was not allocated by the implementation.
+    AC_MSG_CHECKING([whether Bitmap_Free() frees the rdpBitmap and its image data])
+    AC_EGREP_CPP([\"2.0.0-dev\"], [
+
+        #include <freerdp/version.h>
+        FREERDP_VERSION_FULL
```
Comment (jmuehlner) [inline configure.ac]: What specifically are you looking for in the header file to indicate that we're on rc1+?

Is it this `FREERDP_VERSION_FULL` string?

Comment (jmuehlner) [reply configure.ac]: I mean, I get that it's the whole section included here, I was just wondering why you chose this in particular.

Comment (mike-jumper) [reply configure.ac]: This is looking for the specific text `"2.0.0-dev"` to detect that we're on rc0.

`AC_EGREP_CPP` passes the given source through the C preprocessor (CPP) and then egrep. The C preprocessor will expand that macro to its defined value, which will match the expression `"2.0.0-dev"` if the version is 2.0.0-rc0.

Comment (mike-jumper) [reply configure.ac]: I guess I actually need to escape the periods in that regex. This will currently match `"200.0-dev"`, too...

Comment (mike-jumper) [reply configure.ac]: There.

Comment (jmuehlner) [reply configure.ac]: Ah, that makes sense.

**Comment by jmuehlner** [comment]
I'm not sure I understand the title of this PR. From the description, it sounds like you want to free the internals for 2.0.0-rc0 and later, whereas the title says earlier. Am I missing something?

**Comment by mike-jumper** [comment]
Nope, you're right - I have the title backwards.

**Comment by mike-jumper** [comment]
...and one of the commit messages. I'll correct that one, too.

**Comment by mike-jumper** [comment]
OK, I think that should be accurate.

**Pull request merged**